<launch>
    <arg name="circle_offsets" default="[0.1, 0.3]" />
    <arg name="t_max" default="2"/>
    <arg name="delta_t" default="0.1"/>

    <arg name="wheelbase" default="0.3"/>
    <arg name="vehicle_width" default="0.3"/>
    <arg name="vehicle_length" default="0.5"/>
    <arg name="base_link_to_center_dist" default="0.1"/>

    <arg name="odom_topic" default="odom"/>
    <arg name="drive_topic" default="drive"/>

    <node name="collision_warning_system" pkg="collision_warning_system" type="collision_warning_system_node" output="screen">
        <rosparam param="circle_offsets" subst_value="True">$(arg circle_offsets)</rosparam>
        <param name="t_max" value="$(arg t_max)"/>
        <param name="delta_t" value="$(arg delta_t)"/>

        <param name="wheelbase" value="$(arg wheelbase)"/>
        <param name="vehicle_width" value="$(arg vehicle_width)"/>
        <param name="vehicle_length" value="$(arg vehicle_length)"/>
        <param name="base_link_to_center_dist" value="$(arg base_link_to_center_dist)"/>

        <remap from="odom" to="$(arg odom_topic)"/>
        <remap from="drive" to="$(arg drive_topic)"/>
    </node>
</launch>